pkg/stack_mgr/stack.go

```diff
@@ -224,33 +234,110 @@ func (s *Stack) Plan(ctx context.Context, waitOptions options.WaitOptions, dryRu
 
-	logrus.Debugf("will use tf bundle found at %s", srcDir)
+	if util.IsLocalstackMode() {
+		module, diag := tfconfig.LoadModule(srcDir)
+		if diag.HasErrors() {
+			return errors.Wrap(err, "There was an issue loading the module")
+		}
 
-	tempFile, err := os.CreateTemp("", "happy_tfe.*.tar.gz")
-	if err != nil {
-		return errors.Wrap(err, "could not create temporary file")
-	}
-	defer os.Remove(tempFile.Name())
-	err = s.dirProcessor.Tarzip(srcDir, tempFile)
-	if err != nil {
-		return err
-	}
+		cmdPath, err := s.executor.LookPath("tflocal")
+		if err != nil {
+			return errors.Wrap(err, "failed to locate tflocal")
+		}
 
-	configVersionId, err := workspace.UploadVersion(srcDir, dryRun)
-	if err != nil {
-		return err
-	}
+		// Clear out any prior state... For now. Every stack has to have its own
 
-	// TODO should be able to use workspace.Run() here, as workspace.UploadVersion(srcDir)
-	// should have generated a Run containing the Config Version Id
+		// _ = os.Remove(filepath.Join(srcDir, "terraform.tfstate"))
+		// _ = os.Remove(filepath.Join(srcDir, "terraform.tfstate.backup"))
+		_ = os.Remove(filepath.Join(srcDir, "localstack_providers_override.tf"))
 
-	isDestroy := false
-	err = workspace.RunConfigVersion(configVersionId, isDestroy, dryRun)
-	if err != nil {
-		return err
-	}
+		// Run 'terraform init'
 
-	return workspace.WaitWithOptions(ctx, waitOptions, dryRun)
+		cmd := &exec.Cmd{
+			Path:   cmdPath,
+			Args:   []string{"tflocal", "init"},
+			Dir:    srcDir,
+			Stdin:  os.Stdin,
+			Stderr: os.Stderr,
+			Stdout: os.Stdout,
+		}
+		logrus.Infof("%s", cmd.String())
+		logrus.Infof("... in %s", srcDir)
+		if err := s.executor.Run(cmd); err != nil {
+			return errors.Wrap(err, "failed to execute")
+		}
+
+		command := "apply"
+		if bool(dryRun) {
+			command = "plan"
+		}
+		tfArgs := []string{"tflocal", command}
+		if !bool(dryRun) {
+			tfArgs = append(tfArgs, "-auto-approve")
+		}
+
+		// Every stack has to have its own state file.
+		tfArgs = append(tfArgs, fmt.Sprintf("-state=%s.tfstate", s.stackName))
+		tfArgs = append(tfArgs, "-lock=false")
+
+		for param, value := range meta.GetParameters() {
+			if _, ok := module.Variables[param]; ok {
+				tfArgs = append(tfArgs, fmt.Sprintf("-var=%s=%s", param, value))
+			}
+		}
+		metaTags, err := json.Marshal(meta.GetTags())
+		if err != nil {
+			return errors.Wrap(err, "could not marshal json")
+		}
+		if _, ok := module.Variables["happymeta_"]; ok {
+			tfArgs = append(tfArgs, fmt.Sprintf("-var=happymeta_='%s'", string(metaTags)))
+		}
+
+		// Run 'terraform plan' or 'terraform apply'
+
+		cmd = &exec.Cmd{
+			Path:   cmdPath,
+			Args:   tfArgs,
+			Dir:    srcDir,
+			Stdin:  os.Stdin,
+			Stderr: os.Stderr,
+			Stdout: os.Stdout,
+		}
+		logrus.Infof("%s", cmd.String())
+		logrus.Infof("... in %s", srcDir)
+		if err := s.executor.Run(cmd); err != nil {
+			return errors.Wrap(err, "failed to execute")
+		}
+		return nil
+	} else {
```

Comment:
```suggestion
	} 
```
remove the else here its not necessary